src/ow1/heroes/dva.opy

```diff
@@ -51,4 +51,24 @@ rule "[dva.opy]: Cancel booster if melee pressed":
     eventPlayer.forceButtonPress(Button.ABILITY_1)
     eventPlayer.allowButton(Button.MELEE)
     eventPlayer.forceButtonPress(Button.MELEE)
-    eventPlayer.disallowButton(Button.MELEE)
\ No newline at end of file
+    eventPlayer.disallowButton(Button.MELEE)
+
+    
+rule "[dva.opy]: Reduce boosters damage":
+    @Event playerDealtDamage
+    @Hero dva
+    @Condition eventPlayer.isUsingAbility1()
+    @Condition eventDamage == OW2_DVA_BOOSTERS_DAMAGE
+
+    heal(victim,null, OW2_DVA_BOOSTERS_DAMAGE - OW1_DVA_BOOSTERS_DAMAGE)
+
+
+rule "[dva.opy]: Reset Dva health":
+    @Event eachPlayer
+    @Hero dva
+    @Condition eventPlayer.getMaxHealth() == 650
+    @Condition eventPlayer.initialized
+
+    clearCustomHealth()
+    setCustomHealth(OW1_DVA_HEALTH, OW1_DVA_ARMOR, 0)
+    wait()
```

Comment:
I think one problem with this is when DVa takes damage while going back inside Mech. I'm p sure this implementaion heals her back to full hp, but we should test it.

Reply:
Yes it is possible maybe adding a get health before the set custom health and damaging when the health is set to 600 to tr y to reduce the missing damage because I think it will still miss some damage done.